### PR TITLE
109294 Replace hardcoded all_accepted value with plant filter config

### PR DESCRIPTION
**Pull request by haavard-hoijord**
[AB#109294](https://statoildeveloper.visualstudio.com/b6d97460-2e53-47ae-aad4-b41396d90828/_workitems/edit/109294)

### CODE REVIEW

File: FamFeederFunction/Functions/FamFeeder/CutoffForWeekAndPlantHttpTrigger.cs [hunk 1]
```diff
@@ -33,7 +42,8 @@ public class CutoffForWeekAndPlantHttpTrigger
             return new BadRequestObjectResult("Please provide plant");
         }
 
-        MultiPlantConstants.TryGetByMultiPlant("ALL_ACCEPTED", out var enabledPlants);
+        MultiPlantConstants.TryGetByMultiPlant("ALL_ACCEPTED", out var allAccepted);
+        var enabledPlants = new List<string>(_famFeederOptions.PlantFilter?.Replace(" ", "").Split(",") ?? allAccepted.ToArray());
```
Comment (jsoi-eq) [inline FamFeederFunction/Functions/FamFeeder/CutoffForWeekAndPlantHttpTrigger.cs:46]: I'm not sure it makes sense to leave allAccepted as fallback, this will lead to missing data every Monday and we may not discover it.

Suggest throwing an exception instead as this really should be populated.

Could possibly use similar logic to FamWebJob code where we have a separate property PlantFIlterList using PlantFilter property
 public List<string> PlantFilterList =>
     PlantFilter
         ?.Split(",", StringSplitOptions.RemoveEmptyEntries | StringSplitOptions.TrimEntries).ToList()
     ?? throw new Exception(
         $"There is no config value for {nameof(PlantFilter)}. It should be a comma separated list of PCS projectschemas like PCS$GRANE");

Comment (jsoi-eq) [inline FamFeederFunction/Functions/FamFeeder/CutoffForWeekAndPlantHttpTrigger.cs:46]: Would also prefer a unit test to ensure that we are handling trimming of whitespace.

File: Core/FamFeederOptions.cs [hunk 2]
```diff
@@ -8,6 +8,5 @@ public class FamFeederOptions
     public string? SearchIndexBatchSize { get; set; }
     public string? SearchIndexBatchDelay { get; set; }
     public string? DbStatusAiCs { get; set; }
-    
-
+    public string? PlantFilter { get; set; }
```
Comment (jsoi-eq) [inline Core/FamFeederOptions.cs:11]: This file contains a mix of Search stuff and Fam stuff.
Seems like SearchIndex is used as prefix for search.
Suggest using FamFeeder as prefix for Fam stuff, like FamFeederPlantFilter to give context.
Should really be split into two config classes, but that's for another day...

**Review by jsoi-eq — CHANGES_REQUESTED** [review]
Awaiting feedback.

**Review by jsoi-eq — APPROVED** [review]
